packages/jest-matchers/src/utils.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2014, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+// Stringify values and remove circular references
+function stringify(obj: any): string {
+  const set = new Set();
+  return JSON.stringify(obj, (key, value) => {
+    if (typeof value === 'object' && value !== null) {
+      if (set.has(value)) {
```

Comment:
yeah. it's not real, but we probably don't want to print the same object twice :)
maybe it makes sense to check for the nested level